addons/binding/org.openhab.binding.edimax/src/main/java/org/openhab/binding/edimax/internal/commands/AbstractCommand.java

```diff
@@ -0,0 +1,360 @@
+     * GET constructor.
+     */
+    public AbstractCommand() {
+        // default
+    }
+
+    /**
+     * SET constructor.
+     *
+     * @param newValue
+     */
+    public AbstractCommand(T newValue) {
+        setValue = newValue;
+    }
+
+    protected String getCommandString() {
+        List<String> list = getPath();
+        StringBuffer command = new StringBuffer();
+        command.append(XML_HEADER);
+        recurseSubList(list, command);
+        return command.toString();
+    }
+
+    protected void recurseSubList(List<String> list, StringBuffer command) {
+        String element = list.get(0);
+
+        if (list.size() == 1) {
+            // basket case
+            command.append(createLeafTag(element));
+            return;
+        }
+
+        // usually.
+        command.append(createStartTag(element));
+        List<String> subList = list.subList(1, list.size());
+        recurseSubList(subList, command);
+        command.append(createEndTag(element));
+    }
+
+    /**
+     * Overwrite and add your path entry.
+     *
+     * @return
+     */
+    protected List<String> getPath() {
+        ArrayList<String> list = new ArrayList<>();
+        list.add("SMARTPLUG");
+        return list;
+    }
+
+    /**
+     * Returns XPath expression to load response for a get request.
+     *
+     * @return
+     */
+    protected String getXPathString() {
+        List<String> list = getPath();
+        StringBuffer sb = new StringBuffer();
+        for (int i = 0; i < list.size(); i++) {
+            sb.append("/" + list.get(i));
+        }
+        sb.append("/text()");
+        return sb.toString();
+    }
+
+    /**
+     * Format usually used for dates.
+     */
+    private SimpleDateFormat format = new SimpleDateFormat("yyyyMMddkkmmss");
+
+    /**
+     * Extract return types from response.
+     *
+     * @param result
+     * @return
+     */
+    @SuppressWarnings("unchecked")
+    protected T unmarshal(String result) {
+
+        Type tType = extractType();
+
+        if (tType instanceof Class) {
+            Class<?> tempClass = (Class<?>) tType;
+            if (tempClass == Date.class) {
+                try {
+                    return (T) format.parse(result);
+                } catch (ParseException e) {
+                    e.printStackTrace();
+                }
+            } else if (tempClass == Boolean.class) {
+                if ("ON".equals(result)) {
+                    return (T) Boolean.TRUE;
+                } else if ("OFF".equals(result)) {
+                    return (T) Boolean.FALSE;
+                }
+            } else if (tempClass == Integer.class) {
+                return (T) new Integer(result);
+            } else if (tempClass == BigDecimal.class) {
+                return (T) new BigDecimal(result);
+            } else if (tempClass == String.class) {
+                return (T) result;
+            }
+        }
+
+        throw new RuntimeException("Type unknown " + tType);
```

Comment:
Do not throw raw exception types such as the `Exception` or the `RuntimeException`, use a specific exception and if none exists create your own.